### PR TITLE
Flash Attention Implementation & Fuller Config Options

### PR DESCRIPTION
**Pull request by benjaminye**
Addresses https://github.com/georgian-io/LLM-Finetuning-Toolkit/issues/130

### CODE REVIEW

**Review by RohitSaha — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -45,6 +48,30 @@ This command initiates the fine-tuning process using the settings specified in t
...
+
+```shell
+pipx inject llm-toolkit flash-attn --pip-args=--no-build-isolation
+```
+
+**pip**
+
+```
+pip install flash-attn --no-build-isolation
+```
```
Comment (truskovskiyk) [inline README.md:65]: is it possible to install it with llm-toolkit, so users do not have to do this?

File: llmtune/inference/lora.py [hunk 2]
```diff
@@ -64,7 +64,7 @@ def _get_merged_model(self, weights_path: str):
 
         return model, tokenizer
 
-    def infer_all(self):
+    def infer_test_set(self):
```
Comment (truskovskiyk) [inline llmtune/inference/lora.py:67]: do we need to change this in this PR?

**Review by truskovskiyk — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (benjaminye) [reply README.md:65]: Not possible as the package doesn't support PEP 517. See https://github.com/python-poetry/poetry/issues/8427, https://github.com/Dao-AILab/flash-attention/pull/493#issuecomment-1693938950.

File: llmtune/inference/lora.py (hunk 2 above)
Comment (benjaminye) [reply llmtune/inference/lora.py]: Will raise another issue for this.

File: README.md (hunk 1 above)
Comment (benjaminye) [reply README.md:65]: ^^ Also that's the first thing I tried 🙈

**Pull request merged**